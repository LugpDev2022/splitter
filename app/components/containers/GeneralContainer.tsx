interface GeneralContainerProps {
  children: React.ReactNode;
}

const GeneralContainer: React.FC<GeneralContainerProps> = ({ children }) => (
  <div
    className='
    flex
    flex-col
    gap-8
    justify-end
    md:justify-center
    min-h-[1020px]
    md:min-h-0
    h-full
    items-center'
  >
    {children}
  </div>
);

export default GeneralContainer;
