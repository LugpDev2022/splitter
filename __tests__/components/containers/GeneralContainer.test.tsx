import GeneralContainer from '@/app/components/containers/GeneralContainer';
import { render, screen } from '@testing-library/react';

describe('Tests on <GeneralContainer />', () => {
  test('should match snapshot', () => {
    const { container } = render(
      <GeneralContainer>
        <h2>Example h2</h2>
      </GeneralContainer>
    );

    expect(container).toMatchSnapshot();
  });

  test('should render the given content', () => {
    const testText = 'Example h2';

    render(
      <GeneralContainer>
        <h2>{testText}</h2>
      </GeneralContainer>
    );

    expect(screen.getByText(testText)).toBeTruthy();
  });
});
