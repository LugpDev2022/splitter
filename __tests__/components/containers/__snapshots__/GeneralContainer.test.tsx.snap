// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GeneralContainer /> should match snapshot 1`] = `
<div>
  <div
    class=" flex flex-col gap-8 justify-end md:justify-center min-h-[1045px] md:min-h-0 h-full items-center"
  >
    <h2>
      Example h2
    </h2>
  </div>
</div>
`;
